File: examples/Socket/pwmTestS.c
/* Software PWM test
Slowly (over 10 seconds) ramp up brightness of LED then ramp down

TO BUILD

// If building on remote system gcc pwmTestS.c pi-sock.c  -o pwmTestS
// gcc pwmTestS.c -lpi-sock -o pwmTestS

*/
// 2023-10-02

#include <stdio.h>
#include <string.h>
#include <time.h>
#include <unistd.h>

// #include "pi-sock.h"	// If building on remote system 
#include <pi-sock.h>

// #define LEDpin 18
#define LEDpin 12

int main(int argc, char *argv[]) {
  char* host = "127.0.1.1";
  if (argc > 1) { host = argv[1]; } // Optional host address
  if(socket_connect(host)<0)  return -1;
	printf("Ramp up brightness of LED connected to GPIO %d then ramp down\n", LEDpin);

  // ensure channel set as output
  setup_gpio(LEDpin, OUTPUT, 0);
  pwm_set_duty_cycle(LEDpin, 0);
  pwm_set_frequency(LEDpin, 100);
  pwm_start(LEDpin);

  if (pwm_exists(LEDpin))
    printf("pwm enabled on %i\n", LEDpin);

  for (int bright = 0; bright < 100; bright++) {
    usleep(100000);
    pwm_set_duty_cycle(LEDpin, bright);
  }

  for (int bright = 100; bright; bright--) {
    usleep(100000);
    pwm_set_duty_cycle(LEDpin, bright);
  }

  pwm_stop(LEDpin);
  return 0;
}

File: examples/Socket/pwmTestS.py
#! /usr/bin/env python3
"""
Software PWM test
Slowly (over 10 seconds) ramp up brightness of LED then ramp down

Using pi_gpio Socket Interface
"""

# 2023-10-03
import sys
from pi_sock import *
from time import sleep

host = "localhost";
if len(sys.argv) > 1:
  host = sys.argv[1];   # Optional host address

socket_connect(host)

LEDpin = 12

# GPIO.setup()
# print("GPIO.VERSION = {}".format(GPIO.VERSION))

print(f"Ramp up brightness of LED connected to GPIO {LEDpin:d} then ramp down")

setup_gpio(LEDpin, 1, 0)   # Set LEDpin OUTPUT

pwm_set_duty_cycle(LEDpin, 0.0)
pwm_set_frequency(LEDpin, 100.0)
pwm_start(LEDpin);

if (pwm_exists(LEDpin)):
    print(f"pwm enabled on {LEDpin:d}");

for g in range(0, 100, 10):
    sleep(1)
    pwm_set_duty_cycle(LEDpin, float(g))
for g in range(100, -1, -10):
    sleep(1)
    pwm_set_duty_cycle(LEDpin, float(g))

sleep(1)
pwm_stop(LEDpin)
socket_disconnect()
